src/main/java/com/gotocompany/firehose/sink/grpc/client/GrpcClient.java

```diff
@@ -41,39 +45,47 @@ public GrpcClient(FirehoseInstrumentation firehoseInstrumentation, GrpcSinkConfi
         this.managedChannel = managedChannel;
     }
 
-    public DynamicMessage execute(byte[] logMessage, Headers headers) {
-
+    public void initialize() {
         MethodDescriptor.Marshaller<byte[]> marshaller = getMarshaller();
-        DynamicMessage dynamicMessage;
-
-        try {
+        this.methodDescriptor = MethodDescriptor.newBuilder(marshaller, marshaller)
+                .setType(MethodDescriptor.MethodType.UNARY)
+                .setFullMethodName(grpcSinkConfig.getSinkGrpcMethodUrl())
+                .build();
+    }
 
+    public DynamicMessage execute(byte[] logMessage, Headers headers) {
 
+        try {
             Metadata metadata = new Metadata();
             for (Header header : headers) {
                 metadata.put(Metadata.Key.of(header.key(), Metadata.ASCII_STRING_MARSHALLER), new String(header.value()));
             }
-
             Channel decoratedChannel = ClientInterceptors.intercept(managedChannel,
                      MetadataUtils.newAttachHeadersInterceptor(metadata));
             byte[] response = ClientCalls.blockingUnaryCall(
                     decoratedChannel,
-                    MethodDescriptor.newBuilder(marshaller, marshaller)
-                            .setType(MethodDescriptor.MethodType.UNARY)
-                            .setFullMethodName(grpcSinkConfig.getSinkGrpcMethodUrl())
-                            .build(),
-                    CallOptions.DEFAULT,
+                    methodDescriptor,
+                    decorateCallOptions(CallOptions.DEFAULT),
```

Comment:
No need to call this with `CallOptions.DEFAULT`.

Reply:
Agree, made changes.